test/testdrive/mzcompose.py

```diff
@@ -73,11 +79,17 @@ def workflow_default(c: Composition, parser: WorkflowArgumentParser) -> None:
     if args.aws_region is None:
         dependencies += ["localstack"]
 
+    if args.default_timeout is None:
+        timeout_arg = {}
+    else:
+        timeout_arg = {"default_timeout": f"{args.default_timeout}s"}
+
     testdrive = Testdrive(
         forward_buildkite_shard=True,
         kafka_default_partitions=args.kafka_default_partitions,
         aws_region=args.aws_region,
         validate_postgres_stash="materialized",
+        **timeout_arg,
```

Comment:
nit: the optionality should be in `Testdrive.run` (where it's already easy to conditionally add something to the `command` list) and then this can just do `Testdrive(timeout=args.default_timeout)` directly.